Sources/VLCFirstStepsBaseViewController.m

```diff
@@ -59,6 +71,52 @@ - (void)updateTheme
 }
 
 - (void)setupPage
+{
+    [self configurePage];
+    self.pageTitleContainer.hidden = self.isCompactHeight;
+
+    UIView *central = self.centralView;
+    [central removeFromSuperview];
+    [self.view addSubview:central];
+    UIView *bottom = self.bottomView;
+    [bottom removeFromSuperview];
+    [self.view addSubview:bottom];
+
+    id<VLCLayoutAnchorContainer> guide = self.view;
+    if (@available(iOS 11.0, *)) {
+        guide = self.view.safeAreaLayoutGuide;
+    }
+
+    [NSLayoutConstraint deactivateConstraints: self.labelHeightConstraints];
+    [NSLayoutConstraint activateConstraints: self.isCompactHeight ? @[
+        [bottom.leadingAnchor constraintEqualToAnchor:guide.leadingAnchor],
+        [bottom.bottomAnchor constraintGreaterThanOrEqualToAnchor:guide.bottomAnchor],
+        [bottom.widthAnchor constraintEqualToAnchor:guide.heightAnchor],
+        [bottom.topAnchor constraintEqualToAnchor:guide.topAnchor],
+        [central.leadingAnchor constraintEqualToAnchor:bottom.trailingAnchor],
+        [central.bottomAnchor constraintEqualToAnchor:bottom.bottomAnchor],
+        [central.topAnchor constraintEqualToAnchor:guide.topAnchor],
+        [central.trailingAnchor constraintEqualToAnchor:guide.trailingAnchor]
+    ] : @[
+        [bottom.leadingAnchor constraintEqualToAnchor:guide.leadingAnchor],
+        [bottom.bottomAnchor constraintEqualToAnchor:guide.bottomAnchor],
+        [bottom.trailingAnchor constraintEqualToAnchor:guide.trailingAnchor],
+        [central.leadingAnchor constraintEqualToAnchor:bottom.leadingAnchor],
+        [central.bottomAnchor constraintEqualToAnchor:bottom.topAnchor],
+        [central.trailingAnchor constraintEqualToAnchor:bottom.trailingAnchor],
+        [central.topAnchor constraintEqualToAnchor:self.pageTitleContainer.bottomAnchor],
+    ]];
+
+    if (!self.isCompactHeight) {
+        [self.view setNeedsLayout];
```

Comment:
for non-compact height calculate all heights for labels to set the same height (looks better). layout to get correct allowed width for them